src/components/app/app.ts

```diff
@@ -0,0 +1,21 @@
+import AppController from '../controller/controller';
+import { RequestSource } from '../../types/type';
+import { AppView } from '../view/appView';
+
+class App {
+    controller: AppController;
+    view: AppView;
+    constructor() {
+        this.controller = new AppController();
+        this.view = new AppView();
+    }
+
+    start() {
+        (<HTMLTemplateElement>document.querySelector('.sources')).addEventListener('click', (e) =>
```

Comment:
Why do you use HTMLTemplateElement instead of HTMLElement here? it seems the markup is already rendered